tests/utils.go

```diff
@@ -355,9 +355,17 @@ func (w *ObjectEventWatcher) Watch(abortChan chan struct{}, processFunc ProcessF
 	go func() {
 		defer GinkgoRecover()
 		for obj := range eventWatcher.ResultChan() {
-			if f(obj.Object.(*k8sv1.Event)) {
-				close(done)
-				break
+			switch t := obj.Object.(type) {
+			case *k8sv1.Event:
+				if f(obj.Object.(*k8sv1.Event)) {
+					close(done)
+					break
+				}
+			case *metav1.Status:
+				status := obj.Object.(*metav1.Status)
+				Fail(fmt.Sprintf("WARNING: received metav1.Status instead of k8sv1.Event with content %+v", status))
```

Comment:
I'm not quite sure if that's the right approach here, as we are still then having a flaky test here. The test test_id:1511 under question is located in [networkpolicy_test.go](https://github.com/kubevirt/kubevirt/blob/c00e33c2c0ca003ecfd50d343bbbf094ce99d8a1/tests/networkpolicy_test.go#L90), while the actual callers of the method are located solely in storage_test.go and vmi_lifecycle_test.go.

So this sounds like a leftover from one of those tests, which may indicate that we will never see this occur when running this test alone.

WDYT?

Reply:
If we know why a status is returned, we can decide if it is save to ignore it, if we reconnect and so forth. So firt I would assume that we have to understand what we see.

Reply:
I totally agree on that we have to understand what is going on here. So ok, let's proceed with adding a failure in case this happens.

Reply:
Can we experiment here and try to reproduce it by pulling one lane? (It more often fails then not, right?)